public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -123,21 +156,121 @@
+                public org.pmiops.workbench.model.AchillesResult apply(AchillesResult o) {
+
+                    return new org.pmiops.workbench.model.AchillesResult()
+                            .id(o.getId())
+                            .analysisId(o.getAnalysisId())
+                            .stratum1(o.getStratum1())
+                            .stratum2(o.getStratum2())
+                            .stratum3(o.getStratum3())
+                            .stratum4(o.getStratum4())
+                            .stratum5(o.getStratum5())
+                            .stratum5Name(o.getStratum5Name())
+                            .countValue(o.getCountValue());
                 }
             };
 
 
+    /* Final api functions put on top */
+    @Override
+    public ResponseEntity<DbDomainListResponse> getDomainFilters() {
+        List<DbDomain> domains=dbDomainDao.findByDbType("domain_filter");
+        DbDomainListResponse resp=new DbDomainListResponse();
+        resp.setItems(domains.stream().map(TO_CLIENT_DBDOMAIN).collect(Collectors.toList()));
+        return ResponseEntity.ok(resp);
+    }
+
+    @Override
+    public ResponseEntity<DbDomainListResponse> getSurveyList() {
+        List<DbDomain> domains=dbDomainDao.findByDbTypeAndAndConceptIdNotNull("survey");
+        DbDomainListResponse resp=new DbDomainListResponse();
+        resp.setItems(domains.stream().map(TO_CLIENT_DBDOMAIN).collect(Collectors.toList()));
+        return ResponseEntity.ok(resp);
+    }
+
+
+
+    @Override
+    public ResponseEntity<QuestionConceptListResponse> getSurveyResults(String surveyConceptId) {
+        /* Set up the age and gender names */
+        // Too slow and concept names wrong so we hardcode list
+        // List<Concept> genders = conceptDao.findByConceptClassId("Gender");
+        this.setAgeStratumNameMap();
+        this.setGenderStratumNameMap();
+        long longSurveyConceptId = Long.parseLong(surveyConceptId);
+        List<QuestionConcept> questions = questionConceptDao.findSurveyQuestions(longSurveyConceptId);
+        QuestionConceptListResponse resp = new QuestionConceptListResponse();
+        DbDomain survey = dbDomainDao.findByConceptId(longSurveyConceptId);
+        resp.setSurvey(TO_CLIENT_DBDOMAIN.apply(survey));
+
+        if (!questions.isEmpty()) {
+            for (int x = 0; x < questions.size() - 1; x++)  {
+                QuestionConcept q = questions.get(x);
+                String qid = Long.toString(q.getConceptId());
+
+                /*** Important -- when calling these dao functions in a loop, baceause the analyses share same id ,
+                 * and we are filtering by a property in the related reluslts table, the first results get cached and
+                 * put on all questions. So we have to detach the entity from the session
+                 */
+                List<AchillesAnalysis> analyses = achillesAnalysisDao.findQuestionAnalysisResults(surveyConceptId, qid);
```

Comment:
Don't do queries in a loop... if we needed to do a lookup of N IDs, we should use an in clause instead. But better yet, just merge the SQL for findSurveyQuestions with the SQL for this, so you only issue one query instead of two.

Reply:
:) I'll work a little more on this.  QuestionConcept relates to analysis only through the results. Ideally questionconcept can have analyses array which has the results. I'm not sure how to express that in JPA .